<span>Name : </span><input id="name"/><br/>
<span>Delay : </span><input id="delay"/><br/>
<button id="alarm">Set Alarm</button>
<pre readonly id="output"></pre>

<script>
    let name=document.querySelector('#name');
    let delay=document.querySelector('#delay');
    let alarmId=document.querySelector('#alarm');
    let output=document.querySelector('#output');

    alarmId.addEventListener('click',alarmClick)

    function alarm(name,delay){
        return new Promise((resolve,reject)=>{
            if(delay<0){
                throw new Error(`Delay must not be negative`);
            }
            setTimeout(()=>{
                resolve(`WakeUp!! ${name}`);
            },delay*1000);
        })
    }
    async function alarmClick(){
        
        const fetchResponse=await alarm(name.value,delay.value);//await promise-->return response data directly, if we use fetchResponse here within async
        console.log(fetchResponse);// data
        //If delay <0 got error , which is thrown from alarm() is not handled in this async function
        output.textContent=fetchResponse;
    }
</script>